src/modules/job-info/job_state.c

```diff
@@ -1238,6 +1244,80 @@ void job_state_cb (flux_t *h, flux_msg_handler_t *mh,
     return;
 }
 
+static int parse_annotation (json_t *annotation, flux_jobid_t *id, json_t **aValue)
+{
+    json_t *o;
+
+    if (!json_is_array (annotation))
+        return -1;
+
+    if (!(o = json_array_get (annotation, 0))
+        || !json_is_integer (o))
+        return -1;
+
+    (*id) = json_integer_value (o);
+
+    if (!(o = json_array_get (annotation, 1))
+        || (!json_is_object (o) && !json_is_null (o)))
+        return -1;
+
+    (*aValue) = o;
+    return 0;
+}
+
+static void update_annotations (struct info_ctx *ctx, json_t *annotations)
+{
+    struct job_state_ctx *jsctx = ctx->jsctx;
+    size_t index;
+    json_t *value;
+
+    if (!json_is_array (annotations)) {
+        flux_log (ctx->h, LOG_ERR, "%s: annotations EPROTO", __FUNCTION__);
+        return;
+    }
+
+    json_array_foreach (annotations, index, value) {
+        struct job *job;
+        flux_jobid_t id;
+        json_t *aValue;
+
+        if (parse_annotation (value, &id, &aValue) < 0) {
```

Comment:
How about "job %ju annotation parse error"

Reply:
but I won't know the ID :-), i'll adjust to "annotation parse error", to differentiate from the error above it